pkg/metrics/statsd/statsd_test.go

```diff
@@ -83,3 +86,247 @@ func TestEmitFloat(t *testing.T) {
 		t.Error(m)
 	}
 }
+
+func TestEmitGaugeViaUDP(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	env := mock_env.NewMockInterface(controller)
+	env.EXPECT().Location().AnyTimes().Return("eastus")
+	env.EXPECT().Hostname().AnyTimes().Return("test-host")
+
+	socket := "127.0.0.1:8001"
+
+	os.Setenv(statsdSocketEnv, "udp:"+socket)
+	c2, _ := net.ListenPacket("udp", socket)
+
+	s := &statsd{
+		env:       env,
+		account:   "*",
+		namespace: "*",
+		ch:        make(chan *metric),
+		now:       func() time.Time { return time.Time{} },
+	}
+
+	c := make(chan string)
+	go func() {
+		buf := make([]byte, 1024)
+		// set 5 second read timeout. That should be plenty of time to receive the emitted Gauge
+		c2.SetReadDeadline(time.Now().Add(5 * time.Second))
+		m := ""
+		n, _, err := c2.ReadFrom(buf)
+		if err != nil {
+			m = err.Error()
+		} else {
+			m = string(buf[:n])
+		}
+
+		c <- m
+	}()
+
+	go s.run()
+	s.EmitGauge("tests.test_key", 42, map[string]string{"key": "value"})
+	m := <-c
+	if m != `{"Metric":"tests.test_key","Account":"*","Namespace":"*","Dims":{"hostname":"test-host","key":"value","location":"eastus"},"TS":"0001-01-01T00:00:00.000"}:42|g`+"\n" {
+		t.Error(errors.New(m))
+	}
+}
+
+func TestGetConnectionDetails(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	for _, tt := range []struct {
+		name         string
+		set          bool
+		isLocalDev   bool
+		socketstring string
+		network      string
+		address      string
+		wantError    bool
+	}{
+		{
+			name:      "Old Bevaviour / production mode",
+			set:       false,
+			network:   "unix",
+			address:   "/var/etw/mdm_statsd.socket",
+			wantError: false,
+		},
+		{
+			name:       "Old Bevaviour / localdev mode",
+			set:        false,
+			isLocalDev: true,
+			network:    "unix",
+			address:    "mdm_statsd.socket",
+			wantError:  false,
+		},
+		{
+			name:         "Valid UDP env variable",
+			set:          true,
+			socketstring: "udp:127.0.0.1:9000",
+			network:      "udp",
+			address:      "127.0.0.1:9000",
+			wantError:    false,
+		},
```

Comment:
I would also add a case when `isLocalDev: true`, but `socketstring` is set to make sure that we do not override in dev.